build-tools/config/webpack/webpack.prod.conf.js

```diff
@@ -0,0 +1,165 @@
+          {
+            loader: 'url-loader',
+            options: {
+              limit: 10000,
+              name: path.posix.join(config.build.versionPath, 'font/[name].[hash:7].[ext]'),
+            },
+          },
+        ],
+      },
+    ],
+  },
+  devtool: false,
+  output: {
+    filename: path.posix.join('', config.build.versionPath + '/js/[name].js'),
+    chunkFilename: path.posix.join('', config.build.versionPath + '/js/[id].js'),
+    publicPath: config.build.publicPath,
+  },
+  plugins: [
+    new WebpackCleanupPlugin(),
+    new webpack.optimize.ModuleConcatenationPlugin(),
+    new webpack.DefinePlugin({
+      'process.env': env,
+    }),
+    new webpack.NamedChunksPlugin(),
+    new webpack.optimize.UglifyJsPlugin({
+      compress: {
+        warnings: false,
+      },
+      output: {
+        comments: false,
+      },
+    }),
+    new OptimizeCssAssetsPlugin({
+      cssProcessorOptions: {
+        safe: true,
+      },
+    }),
+    new ExtractTextPlugin({
+      filename: path.posix.join(config.build.versionPath, '/css/[name].css'),
+    }),
+    new HtmlWebpackPlugin({
+      filename: config.build.index,
+      template: 'index.html',
+      inject: true,
+      version: config.build.versionPath,
+      minify: {
+        removeComments: true,
+        collapseWhitespace: true,
+        removeAttributeQuotes: false,
+      },
+      chunksSortMode: 'dependency',
+    }),
+    new ImageminPlugin({
+      disable: !config.build.enableImageOptimization,
+      svgo: null,
+      gifsicle: null,
+      pngquant: config.build.enablePNGQuant ? { quality: '65' } : null,
+    }),
+    // split vendor js into its own file
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendor',
+      minChunks: function(module, count) {
+        // any required modules inside node_modules are extracted to vendor
+        return (
+          module.resource &&
+          /\.js$/.test(module.resource) &&
+          module.resource.indexOf(path.join(__dirname, '../../../node_modules')) === 0
+        );
+      },
+    }),
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'manifest',
+      minChunks: Infinity,
+    }),
+    new CopyWebpackPlugin([
+      {
+        from: 'static',
+        to: config.build.versionPath + '/static',
+        ignore: ['.*'],
+      },
+    ]),
+    new CopyWebpackPlugin([
+      {
+        from: 'staticRoot',
+        to: '',
+        ignore: ['.*'],
+      },
+    ]),
+    new BundleAnalyzerPlugin({
+      analyzerMode: 'disabled',
+      generateStatsFile: true,
+      statsFilename: '../stats.json',
+    }),
+    ...(config.build.generateIcons ? (
+    [
+      new FaviconsWebpackPlugin({
+        logo: path.resolve(__dirname, '../../../static/image/favicon.png'),
+        prefix: config.build.versionPath + 'static/favicon/',
+        emitStats: false,
+        persistentCache: false,
+        inject: true,
+        background: '#fff',
+        title: '',
+        icons: {
```

Comment:
Iconset, should be easily be overridden by a config.

Reply:
I don't want to move to much to the config. The config should only contain high level settings. If you want to make an advanced change you can go to the webpack config and change it there.